### PR TITLE
Allow edit entry without changing class type

### LINKED ISSUE #52: As a user, I can edit the start and end date of entries

### PR DESCRIPTION
**Pull request by tjonganthony**
Fix #52 

Main changes:
- EditCommand can be supplied start and end time now.
- New Exception class to be thrown when user tried to convert floating task/deadline to event.

### CODE REVIEW

File: src/main/java/seedu/address/model/task/UniqueTaskList.java [hunk 1]
```diff
@@ -35,6 +37,19 @@ protected DuplicateTaskException() {
      * there is no such matching task in the list.
      */
     public static class EntryNotFoundException extends Exception {}
+    
+    /**
+     * Signals that an operation would have violated the 'no duplicates' property of the list.
+     */
+    public static class EntryConversionException extends DataConversionException {
```
Comment (joeleba) [inline src/main/java/seedu/address/model/task/UniqueTaskList.java:44]: indentation for this method

File: src/main/java/seedu/address/model/task/UniqueTaskList.java [hunk 2]
```diff
@@ -105,6 +120,60 @@ public void updateTitle(Entry toEdit, Title newTitle) throws DuplicateTaskExcept
...
+     */
+    public void updateStartTime(Entry toEdit, LocalDateTime newStartTime) throws EntryNotFoundException, EntryConversionException {
+        assert toEdit != null;
+
+        if (!contains(toEdit)) {
+            throw new EntryNotFoundException();
+        }
+
+    	if (newStartTime != null) {
+    		if (toEdit instanceof Event) {
```
Comment (joeleba) [inline src/main/java/seedu/address/model/task/UniqueTaskList.java]: indentation

File: src/main/java/seedu/address/model/task/UniqueTaskList.java [hunk 3]
```diff
@@ -105,6 +120,60 @@ public void updateTitle(Entry toEdit, Title newTitle) throws DuplicateTaskExcept
...
+     *          if the Entry to be edited cannot be found
+     */
+    public void updateEndTime(Entry toEdit, LocalDateTime newEndTime) throws EntryNotFoundException {
+        assert toEdit != null;
+        if (!contains(toEdit)) {
+            throw new EntryNotFoundException();
+        }
+        
+        if (newEndTime != null) {
+        	if (toEdit instanceof Task) {
```
Comment (joeleba) [inline src/main/java/seedu/address/model/task/UniqueTaskList.java]: indentation

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 4]
```diff
@@ -74,6 +81,8 @@ public CommandResult execute() {
             assert false : "The target entry cannot be missing";
         } catch (DuplicateTaskException e) {
             return new CommandResult(MESSAGE_DUPLICATE_TASK);
+        } catch (EntryConversionException e) {
+        	return new CommandResult(MESSAGE_ENTRY_CONVERSION);
```
Comment (joeleba) [inline src/main/java/seedu/address/logic/commands/EditCommand.java]: indentation

**Review by joeleba — CHANGES_REQUESTED** [review]
Looks good. 1 test failing (but this one is resolved on master). Just fix the indentations then merge in 👍

File: src/main/java/seedu/address/model/task/UniqueTaskList.java (hunk 1 above)
Comment (tjonganthony) [reply src/main/java/seedu/address/model/task/UniqueTaskList.java:44]: done.

File: src/main/java/seedu/address/model/task/UniqueTaskList.java (hunk 2 above)
Comment (tjonganthony) [reply src/main/java/seedu/address/model/task/UniqueTaskList.java]: done.

File: src/main/java/seedu/address/model/task/UniqueTaskList.java (hunk 3 above)
Comment (tjonganthony) [reply src/main/java/seedu/address/model/task/UniqueTaskList.java]: done.

File: src/main/java/seedu/address/logic/commands/EditCommand.java (hunk 4 above)
Comment (tjonganthony) [reply src/main/java/seedu/address/logic/commands/EditCommand.java]: done.

**Review by joeleba — APPROVED** [review]

**Pull request merged**